src/lp/ensure_integer_bounds.jl

```diff
@@ -0,0 +1,12 @@
+function ensure_integer_bounds!(ps::PresolveData, j::Int)
+    # Column was already removed.
+    ps.colflag[j] || return nothing
+
+    if (ps.var_types[j] == BINARY)
+        ps.lcol[j] = T(max(0, ceil(ps.lcol[j])))
+         ps.ucol[j] = T(min(1, floor(ps.ucol[j])))
+    elseif (ps.var_types[j] == GENERAL_INTEGER)
+        ps.lcol[j], ps.ucol[j] = T(ceil(ps.lcol[j])), T(floor(ps.ucol[j]))
+    end
```

Comment:
We should have some tolerance here.
Currently, if the lower bound is `1e-10` (which may happen after some roundings), it will get bumped to `1`, which seems overly aggressive.

What I would suggest is to have an `ϵ_int` argument, and tighten the bounds only if their current value is "fractional enough" (see the description of `FixedVariableRule` in #7). Otherwise, set the bound to an integer value, e.g., `1e-8` would become `0` (which will be less prone to numerical issues).